### PR TITLE
Upgrade openlayer map to 4.6.4

### PR DESCRIPTION
**Pull request by daniwe4**
Upgrade openlayer map to 4.6.4

### CODE REVIEW

**Comment by chfsx** [comment]
@klees is this correctly assigned to you?
@daniwe4 thanks for moving this to bower!

**Comment by klees** [comment]
@chfsx Yes, indeed.

File: Services/Maps/css/service_openlayers.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    width: 100% !important;
+}
+.map:-ms-fullscreen {
+    height: 100% !important;
+    width: 100% !important;
+}
+.map:fullscreen {
+    height: 100% !important;
+    width: 100% !important;
+}
```
Comment (klees) [inline Services/Maps/css/service_openlayers.css:16]: Why are these required?

File: Services/Maps/js/ServiceOpenLayers.js [hunk 2]
```diff
@@ -1,434 +1,387 @@
...
-		jQuery("#" + id + "_lng").val(userPos.lon);
-
-		if (map.__replace_marker__) {
-			this.setCentralMarker(id, pos);
+/**
+ * Holds all functionality to work with openlayer maps.
+ *
+ * @author Daniel Weise <daniel.weise@concepts-and-training.de>
+ */
+ServiceOpenLayers = {
```
Comment (klees) [inline Services/Maps/js/ServiceOpenLayers.js]: Lets move this complete initialisation into an anonymus function that is called verbatim, so we don't pollute the global namespace.

File: Services/Maps/js/ServiceOpenLayers.js [hunk 3]
```diff
@@ -1,434 +1,387 @@
...
+	 *
+	 * @param 	{object} 	ol
+	 * @param 	{object} 	jQuery
+	 * @param 	{array} 	map_data
+	 * @param 	{string} 	invalid_address
+	 * @param 	{array} 	user_markers
+	 * @return 	{object}
+	 */
+	create: function(ol, jQuery, map_data, invalid_address, user_markers) {
+		var obj = Object.create(this);
```
Comment (klees) [inline Services/Maps/js/ServiceOpenLayers.js:24]: What is it good for? Isn't `this` already an object?

File: Services/Maps/js/ServiceOpenLayers.js [hunk 4]
```diff
@@ -1,434 +1,387 @@
...
-		feature.popup = feature.createPopup(true);
-		map.addPopup(feature.popup);
-		feature.popup.hide();
-		marker.marker.events.register("click", feature, function(ev) {
-			OpenLayers.Event.stop(ev);
-			this.popup.toggle();
+				lon = parseFloat(data[0].lon, 10);
+				lat = parseFloat(data[0].lat, 10);
+
+				pos = module.posToOSM([lon, lat]);
```
Comment (klees) [inline Services/Maps/js/ServiceOpenLayers.js]: Lets not pollute the global namespace here.

File: Services/Maps/js/ServiceOpenLayers.js [hunk 5]
```diff
@@ -1,434 +1,387 @@
...
+		element.innerHTML = "<img src='./Services/Maps/images/mm_20_blue.png'>";
+	},
+
+	/**
+	 * Remove all child elements.
+	 *
+	 * @returns 	{void}
+	 */
+	deleteAllMarkers: function() {
+		marker = document.getElementsByClassName('marker');
```
Comment (klees) [inline Services/Maps/js/ServiceOpenLayers.js]: Lets not polute the global namespace here.

File: Services/Maps/js/ServiceOpenLayers.js [hunk 6]
```diff
@@ -1,434 +1,387 @@
...
+		this.map.addOverlay(popup);
+	},
+
+	/**
+	 * Delete all popups with class arrow_box.
+	 *
+	 * @returns 	{void}
+	 */
+	deleteAllPopups: function() {
+		popups = document.getElementsByClassName('arrow_box');
```
Comment (klees) [inline Services/Maps/js/ServiceOpenLayers.js]: Lets not pollute the global namespace here.

**Review by klees — COMMENTED** [review]
LGTM in general, please fix the namespace pollution. Thx!

File: Services/Maps/js/ServiceOpenLayers.js (hunk 2 above)
Comment (daniwe4) [reply Services/Maps/js/ServiceOpenLayers.js]: add an var before the object name, so i think it doesn´t pollute the global namespace anymore

File: Services/Maps/js/ServiceOpenLayers.js (hunk 3 above)
Comment (daniwe4) [reply Services/Maps/js/ServiceOpenLayers.js:24]: its like an constructor

File: Services/Maps/css/service_openlayers.css (hunk 1 above)
Comment (daniwe4) [reply Services/Maps/css/service_openlayers.css:16]: without these css snippets the size of the map didn´t change and the rest of the page is only filled black. The same goes for the important tag.

File: Services/Maps/classes/class.ilOpenLayersMapGUI.php [hunk 7]
```diff
@@ -78,7 +78,9 @@ function getHtml()
 
 
 		$lng->loadLanguageModule("maps");
-		$tpl->addJavaScript("Services/Maps/js/OpenLayers.js");
+		$tpl->addCss("libs/bower/bower_components/OpenLayers/ol.css");
+		$tpl->addJavaScript("libs/bower/bower_components/OpenLayers/ol.js");
```
Comment (klees) [inline Services/Maps/classes/class.ilOpenLayersMapGUI.php:82]: These files are missing in the diff. The bower-deps need to be added to the repo just like the composer deps. It is not sufficient to just update the bower.json since we are shipping all code via this repo.

**Review by klees — CHANGES_REQUESTED** [review]

File: Services/Maps/classes/class.ilOpenLayersMapGUI.php (hunk 7 above)
Comment (daniwe4) [reply Services/Maps/classes/class.ilOpenLayersMapGUI.php:82]: done
